query/src/sql/planner/binder/join.rs

```diff
@@ -229,7 +231,13 @@ impl<'a> JoinConditionResolver<'a> {
                 },
             );
 
-            // TODO(leiysky): bump types of left conditions and right conditions
+            // Bump types of left conditions and right conditions
+            let left_type = left.data_type();
+            let right_type = right.data_type();
+            let least_super_type = merge_types(&left_type, &right_type)?;
```

Comment:
@sundy-li Seems `compare_coercion` doesn't work with `Nullable` types, is that by design?

By the way, it looks have the same function with `merge_types` except the nullability check.

Reply:
Yes， it's by design.  I'll document it that we should use `merge_types` if we should care about nullable.

Reply:
> Yes， it's by design.  I'll document it that we should use `merge_types` if we should care about nullable.

IMHO, `get_least_super_type` is a reasonable name.